### PR TITLE
Re-add referrals claim

### PR DESCRIPTION
**Pull request by dospore**
Add referrals claim

### CODE REVIEW

File: src/views/Referrals/Referrals.js [hunk 1]
```diff
@@ -124,13 +138,61 @@ export default function Referral(props) {
...
+
   function handleClaim() {
-    // TODO handle claim
+    console.log("Claiming rewards");
+    setIsClaiming(true);
+    // helperToast.error("Claiming rewards is currently disabled");
+    trackAction("Button clicked", {
+      buttonName: "Claim rewards",
+    });
+    console.log(userProof);
```
Comment (doylio) [inline src/views/Referrals/Referrals.js]: Leftover console.logs

File: src/views/Referrals/Referrals.js [hunk 2]
```diff
@@ -124,13 +138,61 @@ export default function Referral(props) {
...
+    let error;
+    if (selectedRound === "latest") {
+      helperToast.error("Cannot claim rewards before round has ended");
+      error = true;
+    }
+    if (!userProof) {
+      helperToast.error("Fetching merkle proof");
```
Comment (doylio) [inline src/views/Referrals/Referrals.js]: This error message could be a bit more helpful.  Maybe "Fetching merkle proof.  Please wait a minute and try again"

**Review by doylio — CHANGES_REQUESTED** [review]

File: src/views/Referrals/Referrals.js [hunk 3]
```diff
@@ -124,13 +138,59 @@ export default function Referral(props) {
...
   };
 
+  const merkleDistributor = getContract(chainId, "MerkleDistributor");
+  const merkleDistributorReader = getContract(chainId, "MerkleDistributorReader");
+
   function handleClaim() {
-    // TODO handle claim
+    setIsClaiming(true);
+    trackAction("Button clicked", {
+      buttonName: "Claim rewards",
```
Comment (chrdevmar) [inline src/views/Referrals/Referrals.js]: should this be Claim referral rewards? I assume claiming trading rewards uses this same thing

Comment (dospore) [reply src/views/Referrals/Referrals.js]: yeah, probably, idk with the analytics tbh

Comment (dospore) [reply src/views/Referrals/Referrals.js]: pretty sure its just a string trait so I just added referral in there

**Review by chrdevmar — APPROVED** [review]
lgtm

**Pull request merged**